waltz-ng/client/data-types/index.js

```diff
@@ -1,5 +1,28 @@
+import _ from "lodash";
+
 export default (module) => {
 
-    module.service('DataTypesDataService', require('./services/data-types-data'));
+    module.run(['$rootScope', '$state', 'DataTypesService', function($rootScope, $state, dataTypesService) {
+        $rootScope.$on('$stateChangeStart', function(evt, to, params) {
```

Comment:
this _sounds_ worryingly broad scoped.....  Does it not apply to all state changes ?